extensions/objects/templates/literally-canvas-diagram-editor.component.ts

```diff
@@ -0,0 +1,102 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for literally canvas diagram editor.
+ */
+
+const LC = require('literallycanvas');
+require('literallycanvas/lib/css/literallycanvas.css');
+require('services/literally-canvas-helper.service.ts');
+
+angular.module('oppia').component('literallyCanvasDiagramEditor', {
+  template: require('./literally-canvas-diagram-editor.component.html'),
+  controller: [
+    'LiterallyCanvasHelperService', function(LiterallyCanvasHelperService) {
+      const ctrl = this;
+      var MAX_DIAGRAM_WIDTH = 491;
+      var MAX_DIAGRAM_HEIGHT = 551;
+      var DEFAULT_STROKE_WIDTH = 2;
+      var STROKE_WIDTH = [1, 2, 3, 5, 30];
+      ctrl.diagramWidth = 450;
+      ctrl.currentDiagramWidth = 450;
+      ctrl.diagramHeight = 350;
+      ctrl.currentDiagramHeight = 350;
+      ctrl.data = {};
+      ctrl.onWidthInputBlur = function() {
+        if (ctrl.diagramWidth < MAX_DIAGRAM_WIDTH) {
+          ctrl.currentDiagramWidth = ctrl.diagramWidth;
+          ctrl.lc.setImageSize(
+            ctrl.currentDiagramWidth, ctrl.currentDiagramHeight);
+        }
+      };
+
+      ctrl.onHeightInputBlur = function() {
+        if (ctrl.diagramHeight < MAX_DIAGRAM_HEIGHT) {
+          ctrl.currentDiagramHeight = ctrl.diagramHeight;
+          ctrl.lc.setImageSize(
+            ctrl.currentDiagramWidth, ctrl.currentDiagramHeight);
+        }
+      };
+
+      ctrl.getDiagramSizeInfo = function() {
+        var maxWidth = MAX_DIAGRAM_WIDTH;
+        var maxHeight = MAX_DIAGRAM_HEIGHT;
+        return 'This diagram has a maximum dimension of ' + maxWidth +
+        'px X ' + maxHeight + 'px to ensure that it fits in the card.';
+      };
+
+      ctrl.validate = function(data) {
+        // Will be implemented once data is saved.
+        return false;
+      };
+
+      ctrl.$onInit = function() {
+        LC.defineSVGRenderer(
+          'Rectangle', LiterallyCanvasHelperService.rectangleSVGRenderer);
+        LC.defineSVGRenderer(
+          'Ellipse', LiterallyCanvasHelperService.ellipseSVGRenderer);
+        LC.defineSVGRenderer(
+          'Line', LiterallyCanvasHelperService.lineSVGRenderer);
+        LC.defineSVGRenderer(
+          'LinePath', LiterallyCanvasHelperService.linepathSVGRenderer);
+        LC.defineSVGRenderer(
+          'Polygon', LiterallyCanvasHelperService.polygonSVGRenderer);
+        LC.defineSVGRenderer(
+          'Text', LiterallyCanvasHelperService.textSVGRenderer);
+        ctrl.lc = LC.init(document.getElementById('lc'), {
+          imageSize: {
+            width: ctrl.diagramWidth, height: ctrl.diagramHeight},
+          imageURLPrefix: '/assets/literallyCanvas/img',
+          toolbarPosition: 'bottom',
+          defaultStrokeWidth: DEFAULT_STROKE_WIDTH,
+          strokeWidths: STROKE_WIDTH,
+          // Eraser tool is removed because svgRenderer has not been
+          // implimented in LiterallyCanvas. Can include once it is
```

Comment:
"implemented" is misspelled, here and in the next line.

Reply:
Done.